src/main/java/com/tiff/tffnserachservice/controller/TiffnerController.java

```diff
@@ -0,0 +1,64 @@
+package com.tiff.tffnserachservice.controller;
+
+
+import com.tiff.tffnserachservice.dto.TiffnerDTO;
+import com.tiff.tffnserachservice.model.Tiffner;
+import com.tiff.tffnserachservice.repository.TiffnerRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/tffn-search")
+public class TiffnerController {
+
+    @Autowired
+    private TiffnerRepository tiffnerRepository;
+
+    @GetMapping
+    public List<Tiffner> findTiffns() {
+        return tiffnerRepository.findAll();
+    }
+
+    @GetMapping("/api/find-tiffn/{id}")
+    public ResponseEntity<Tiffner> findTffn(@PathVariable Long id) {
+        return tiffnerRepository.findByTiffnerId(id).map(ResponseEntity::ok).orElseThrow();
+    }
+
+
+    @DeleteMapping("/api/delete-tiffn/{id")
+    public Boolean deleteTffn(Long id) {
+
+        if (tiffnerRepository.findByTiffnerId(id).isPresent()) {
```

Comment:
its expensive to call the findByTiffnerId method twice, doing 2 look ups in our db, we can reduce to just 1. Use an optional to store the findBy and use that to determine if present -> delete or return an error